### ISSUE TITLE
$eval incorrectly renders boolean values when there's an error

### ISSUE THREAD

**Issue by SomeKittens** [angular/angular.js#12089]
In 1.4.0, with `a.b` being `undefined` (thus triggering an error):

```
{{ true && a.b.c }} => 'true'
{{ false || a.b.c }} => 'false'
```

Plunkr: http://plnkr.co/edit/Z0OYWcLgqEa0Hil9RyGd?p=preview

This works as expected in 1.3.15: http://plnkr.co/edit/WQ15uAjNYdKEmpeQwvQT?p=preview

**Comment by SomeKittens** [comment]
What's especially troubling is the first case (resulting in true) as this caused all sorts of havoc with our `ng-if` and `ng-show` statements.

**Comment by steelstyle** [comment]
Seems that is is also order dependent

http://plnkr.co/edit/p2J36xikdG9iA6vJw0Bk?p=preview

**Comment by gkalpak** [comment]
Closing as duplicate of #11959.
(This has been fixed on master by #11989 and the fix will be available in `v1.4.1`.)
